modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java

```diff
@@ -51,21 +52,10 @@
 	public void tearDown() throws Exception {
 		_serviceRegistration.unregister();
 
-		try {
-			_configuration.delete();
-		}
-		catch (IllegalStateException ise) {
+		Object delegatee = ReflectionTestUtil.getFieldValue(
+			_configuration, "delegatee");
 
-			// There is a bug in the ConfigurationImpl class where if the
-			// persistence delete method throws, the configuration will still
-			// contain an internal field called 'deleted' that is set to true.
-			// This causes the configuration to throw an IllegalStateException
-			// if any other persistence methods are called on it afterwards.
-			// If the 'onBeforeDelete' method throws, this state occurs. To
-			// work around it for this test, we are catching the
-			// IllegalStateException during clean up.
-
-		}
+		ReflectionTestUtil.invoke(delegatee, "delete", new Class<?>[0]);
```

Comment:
This leads to the question of how do we truly deal with this in an actual use-case and not just in a test?  I'm not sure of the answer.  I sent a [pull request](https://github.com/apache/felix/pull/113) to Apache Felix, but even if they merge it I'm not sure what that process looks like for us to use that.

Is there any way to work around this issue?  Or is it just a bug in waiting?